src/main/java/com/epam/aidial/core/service/PublicationService.java

```diff
@@ -363,7 +367,8 @@ private void prepareAndValidatePublicationRequest(ProxyContext context, String b
                 .toList();
 
         // validate if user has access to all target resources
-        boolean hasPublicAccess = accessService.hasPublicAccess(targetResources, context);
+        boolean hasPublicAccess = targetResources.stream()
```

Comment:
Why did you change the access check from public access to has read access?

Reply:
Changed.

Reply:
previous version was optimized, this one - is not, please use optimized version because permission check for public resources quite expensive

Reply:
Done.